models.py

```diff
@@ -284,8 +285,10 @@ def forward(
 	) -> shaping.BCt:
 
 		if self.frontend is not None:
-			mask = temporal_mask(x, compute_output_lengths(x, xlen)) if xlen is not None else None
-			x = self.frontend(x.squeeze(1), mask = mask)
+			ol = compute_output_lengths(x, xlen)
```

Comment:
for simplicity, can this line go after the squeeze? so that it's clear that it does not depend on being called on not-squeezed input

Reply:
done